moodle/Sniffs/Files/BoilerplateCommentSniff.php

```diff
@@ -64,53 +64,131 @@ public function process(File $file, $stackptr) {
 
         $tokens = $file->getTokens();
 
+        // Allow declare() call in same line as opening tag.
+        if (!strpos($tokens[$stackptr]['content'], "\n") === 0) {
+            if (array_key_exists($stackptr + 1, $tokens) && $tokens[$stackptr + 1]['code'] === T_DECLARE) {
+                // Don't use findEndOfStatement() as a declare statement can legally be followed directly by an expression.
+                $stackptr = $tokens[$stackptr + 1]['parenthesis_closer'];
+
+                // Invalid syntax if nothing after declare.
+                if ($stackptr === count($tokens) - 1) {
+                    return;
+                }
+
+                $stackptr++;
+
+                // Allow multi-line whitespace as it will be picked up by SemicolonSpacingSniff.
+                while ($tokens[$stackptr]['code'] === T_WHITESPACE && array_key_exists($stackptr + 1, $tokens)) {
+                    $stackptr++;
+                }
+
+                if ($tokens[$stackptr]['code'] !== T_SEMICOLON) {
+                    $file->addError(
+                        'First line declare may only be followed by whitespace and semicolon',
+                        $stackptr,
+                        'InvalidDeclare'
+                    );
+                    return;
+                }
+
+                if ($stackptr < count($tokens) - 1) {
+                    $stackptr++;
+
+                    if ($tokens[$stackptr]['code'] !== T_WHITESPACE || $tokens[$stackptr]['content'] !== "\n") {
+                        $file->addError(
+                            'Declare statement may only be followed by a new line',
+                            $stackptr,
+                            'InvalidTrailingCode'
+                        );
+
+                        return;
+                    }
+                }
+            }
+        }
+
         // Allow T_PHPCS_XXX comment annotations in the first line (skip them).
         if ($commentptr = $file->findNext(Tokens::$phpcsCommentTokens, $stackptr + 1, $stackptr + 3)) {
             $stackptr = $commentptr;
         }
 
-        // Find count the number of newlines after the opening <?PHP. We only
-        // count enough to see if the number is right.
-        // Note that the opening PHP tag includes one newline.
-        $numnewlines = 0;
-        for ($i = $stackptr + 1; $i <= $stackptr + 5; ++$i) {
```

Comment:
There's a few ways of doing this... see https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/src/Standards/PEAR/Sniffs/Commenting/FileCommentSniff.php#L95-L104 for an example